### PR TITLE
Generic message option

### PR DESCRIPTION
**Pull request by jjb**
also contains code from https://github.com/zombocom/rack-timeout/pull/205, which could perhaps get merged first

full disclosure, I haven't tried this yet in my own apps

### CODE REVIEW

File: lib/rack/timeout/core.rb [hunk 1]
```diff
@@ -133,17 +135,22 @@ def call(env)
       timeout = RT::Scheduler::Timeout.new do |app_thread|  # creates a timeout instance responsible for timing out the request. the given block runs if timed out
         register_state_change.call :timed_out
 
-        message = "Request "
-        message << "waited #{info.ms(:wait)}, then " if info.wait
-        message << "ran for longer than #{info.ms(:timeout)} "
+        if generic_message
+          message = 'Request timed out'
```
Comment (dentarg) [inline lib/rack/timeout/core.rb:139]: Idea, let the user supply the message?

**Comment by dentarg** [comment]
IMHO "fixed" is a better word over "generic" – but better yet, we can allow the user to set the message :)

**Comment by dentarg** [comment]
It would also be cool to add the info currently residing in the message, as properties on the exception, so even if you need a fixed message, the info (timings) are not lost. WDYT?

**Comment by wuputah** [comment]
hello, thanks for the PR! I don't foresee adding this in this way (via an option), but perhaps:
* the custom exception throwing [suggestion I made](https://github.com/zombocom/rack-timeout/pull/198#issuecomment-2111402344) could be implemented, which would allow you to customize the exception message as you see fit
* this can be fixed via your exception tracking tool, e.g. see [doc/rollbar.md](https://github.com/zombocom/rack-timeout/blob/main/doc/rollbar.md)

**Pull request closed without merging**